Core/include/Acts/Definitions/Algebra.hpp

```diff
@@ -22,71 +22,75 @@
 
 namespace Acts {
 
-/// This is the Scalar used for all EDM and Geometry based object,
-/// can be customised.
+/// @defgroup acts-algebra-types Vector/matrix types with a common scalar type
+///
+/// These are the default vector/matrix types that should be used throughout the
+/// codebase. They all use the common Acts scalar type but support variable size
+/// either at compile- or runtime.
+///
+/// Eigen does not have a distinct type for symmetric matrices. A typedef for
+/// fixed-size matrices is still defined to simplify definition (one template
+/// size vs two template size for generic matrices) and to clarify semantic
+/// meaning in interfaces. It also ensures that the matrix is square. However,
+/// the user is responsible for ensuring that the values are symmetric.
+///
+/// Without a distinct type for symmetric matrices, there is no way to provide
+/// any conditions e.g. square size, for the dynamic-sized case. Consequently,
+/// no dynamic-sized symmetric matrix type is defined. Use the
+/// `ActsDynamicMatrix` instead.
+///
+/// @{
+
+/// Common scalar (floating point type used for the default algebra types.
+///
+/// Defaults to `double` but can be customized by the user.
 #ifdef ACTS_CUSTOM_SCALARTYPE
 using ActsScalar = ACTS_CUSTOM_SCALARTYPE;
 #else
 using ActsScalar = double;
 #endif
 
-template <typename T, unsigned int rows, unsigned int cols>
-using ActsMatrix = Eigen::Matrix<T, rows, cols>;
+template <unsigned int kSize>
```

Comment:
As per your code guidelines, shouldn't this be something like `vector_size_t`?

Reply:
We have always applied *our* [naming guideline N.5](https://acts.readthedocs.io/en/latest/codeguide.html#n-5-constant-values-use-kcamelcase), i.e. template value parameters are treated as constants. [N.7](https://acts.readthedocs.io/en/latest/codeguide.html#n-7-template-parameter-types-use-snake-case-t) only applies to template type parameters.

I think we previously had a discussion on the same topic and decided to follow the [Google Styleguide (bottom paragraph)](https://google.github.io/styleguide/cppguide.html#General_Naming_Rules) on this. Thus the `k...` naming style.

Reply:
You're right. Ok!